tests/resources/test_object_setters.py

```diff
@@ -59,7 +59,7 @@ def test_object_validation():
 
 def test_list_validation():
     """Test that lists are validated by gemd."""
-    with pytest.raises(TypeError):
+    with pytest.raises(ValueError):
```

Comment:
I totally agree with you, this should be a TypeError.
However, because we're now going through every element in the list and trying to deserialize according to the underlying type, this error is now coming from here: https://github.com/CitrineInformatics/citrine-python/blob/PLA-7068/set-list-flexibility/src/citrine/_serialization/properties.py#L71
Which raises a ValueError with the message `'{} is not one of valid types:`. Which is wrong. I opted not to fix this in this PR cause it affects a bunch of other tests. We can fix it in a follow-up if that's okay?

Reply:
v. reasonable